packages/js/product-editor/src/components/variations-table/variation-actions-menus/variation-quick-update-menu-item.tsx

```diff
@@ -0,0 +1,110 @@
+/**
+ * External dependencies
+ */
+import { Slot, Fill, MenuItem } from '@wordpress/components';
+import { createElement, Fragment } from '@wordpress/element';
+import {
+	createOrderedChildren,
+	sortFillsByOrder,
+} from '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import { MenuItemProps } from './types';
+import {
+	MULTIPLE_UPDATE,
+	SINGLE_UPDATE,
+	VARIATION_ACTIONS_SLOT_NAME,
+} from './constants';
+
+const DEFAULT_ORDER = 20;
+
+export const getGroupName = (
+	group?: string,
+	isMultipleSelection?: boolean
+) => {
+	const nameSuffix = isMultipleSelection
+		? `_${ MULTIPLE_UPDATE }`
+		: `_${ SINGLE_UPDATE }`;
+	return group
+		? `${ VARIATION_ACTIONS_SLOT_NAME }_${ group }${ nameSuffix }`
+		: VARIATION_ACTIONS_SLOT_NAME;
+};
+
+export const getMenuItem = (
+	children: React.ReactNode,
+	onClick: () => void
+) => <MenuItem onClick={ onClick }>{ children }</MenuItem>;
+
+export const VariationQuickUpdateMenuItem: React.FC< MenuItemProps > & {
+	Slot: React.FC<
+		Slot.Props & {
+			group: string;
+			supportsMultipleSelection: boolean;
+			onClick?: () => void;
+		}
+	>;
+} = ( {
+	children,
+	order = DEFAULT_ORDER,
+	group = 'top-level',
+	supportsMultipleSelection,
+	onClick = () => {},
+} ) => {
+	if ( supportsMultipleSelection ) {
+		return (
+			<>
+				{ [ MULTIPLE_UPDATE, SINGLE_UPDATE ].map( ( updateType ) => (
+					<Fill
+						key={ updateType }
+						name={ getGroupName(
+							group,
+							updateType === MULTIPLE_UPDATE
+						) }
+					>
+						{ ( fillProps: Fill.Props ) =>
+							createOrderedChildren(
+								getMenuItem( children, onClick ),
+								order,
+								fillProps
+							)
+						}
+					</Fill>
+				) ) }
+			</>
+		);
+	}
+	return (
+		<Fill name={ getGroupName( group, supportsMultipleSelection ) }>
+			{ ( fillProps: Fill.Props ) =>
+				createOrderedChildren(
+					getMenuItem( children, onClick ),
+					order,
+					fillProps
+				)
+			}
+		</Fill>
+	);
```

Comment:
It would be nice to rework this to avoid the code duplication and multiple returns. It's a little confusing to look at, and having the `<Fill>` twice in this function could lead to bugs down the road if a change was made in one path and not the other (say if another prop needed to be passed to the children of the fill.

Reply:
Done [c1d9901](https://github.com/woocommerce/woocommerce/pull/43441/commits/c1d9901ae213acb5071611a5ad2e198a4b4ff30a)